app/src/main/java/org/pragyan/dalal18/ui/MainActivity.kt

```diff
@@ -143,6 +186,34 @@ class MainActivity : AppCompatActivity(), ConnectionUtils.OnNetworkDownHandler {
                             changeTextViewValue(binding.cashWorthTextView, binding.cashIndicatorImageView, cashWorth)
                             changeTextViewValue(binding.totalWorthTextView, binding.totalIndicatorImageView, totalWorth)
                         }
+                        GameStateUpdateType.UserRewardCreditUpdate->{
+
+                            val gameState = intent.getParcelableExtra<GameStateDetails>(GAME_STATE_KEY)
+                            totalWorth += gameState.userRewardCash - cashWorth
+                            cashWorth = gameState.userRewardCash
+                            val reward = gameState.userRewardCash - cashWorth
+                            toast("Congratulations!! You have been rewarded with ${reward} ")
+                            changeTextViewValue(binding.cashWorthTextView, binding.cashIndicatorImageView, cashWorth)
+                            changeTextViewValue(binding.totalWorthTextView, binding.totalIndicatorImageView, totalWorth)
+                        }
+                        GameStateUpdateType.DailyChallengeStatusUpdate->{
+                            val gameState = intent.getParcelableExtra<GameStateDetails>(GAME_STATE_KEY)
+                            val dailychallengeStatus: ImageView = MenuItemCompat.getActionView(binding.navigationViewLeft.menu.findItem(R.id.dailyChallenge_dest)) as ImageView
+                            if(gameState.isDailyChallengeOpen) {
+                                toast("Daily challenge is open")
+
+
+                                dailychallengeStatus.background = getDrawable(R.drawable.orange_status_circle)
+                                dailychallengeStatus.layoutParams.height = 75
+                                dailychallengeStatus.layoutParams.width =75
+
+
+                            }
+                            else {
+                                toast("Daily challenge is closed")
```

Comment:
can we make a marqueue intead of showing a **toast** ?

For referrence - see the main screen when the market is closed.